packages/next/src/client/components/react-dev-overlay/app/ReactDevOverlay.tsx

```diff
@@ -23,27 +22,20 @@ export default class ReactDevOverlay extends React.PureComponent<
   },
   ReactDevOverlayState
 > {
-  state = { reactError: null }
+  state = { isReactError: false }
 
   static getDerivedStateFromError(error: Error): ReactDevOverlayState {
-    if (!error.stack) return { reactError: null }
+    if (!error.stack) return { isReactError: false }
 
     RuntimeErrorHandler.hadRuntimeError = true
     return {
-      reactError: {
```

Comment:
This is not used anymore, I didn't figure out last time how to get rid of this. Some tests were failing when I replace `[reactError]` with `state.errors`.

This time I found that it's the issue with not reporting `error.cause` in `onRecoverableError`, otherwise it's erroring with `"There was an error during concurrent rendering but React was able to recover by instead synchronously rendering the entire root."`. The critical change is in `onRecoverableError`